internal/backend/remote-state/gcs/backend.go

```diff
@@ -212,5 +223,15 @@ func (b *Backend) configure(ctx context.Context) error {
 		b.encryptionKey = k
 	}
 
+	// Customer-managed encryption
+	kmsName := data.Get("kms_encryption_key").(string)
+	if kmsName != "" {
+		kc, err := backend.ReadPathOrContents(kmsName)
```

Comment:
I'm confused by the use of `ReadPathOrContents`, both here and when processing `encryption_key`. From reading the docs I understood `encryption_key` to be an inline base64 encoded key, and `kms_encryption_key` to be a GCP path reference.

`ReadPathOrContents` seems to try to read a local file by treating its argument as a relative path, and only upon failing returns the path. If that functionality is desirable, I think we should document it; if it's not, I think we can just skip calling this function altogether.

Reply:
Yeah I was similarly confused- I chose to use that function to be consistent with `encryption_key`. I'm not keen to remove it from `encryption_key` because it could be a breaking change for some. I'll add it to the docs for both fields!

Reply:
This `ReadPathOrContents` helper is a very old shim for when we decide to standardize on passing settings around in strings rather than using filenames in most cases, so folks would be able to construct the input in various different ways rather than always forcing a file on disk.

However as well as now being very legacy, I'm also skeptical about it being used in a backend configuration context because dynamic expressions aren't allowed here anyway... anything assigned to a backend configuration must always be static.

If we do want to support both literal assignment and assignment by filename then for modern designs I would suggest separate arguments for each and a validation rule preventing using them together. Overloading a single argument to support two different usages was a pragmatic compatibility shim for some existing arguments that already had the name we wanted to keep, but for new designs we can use an argument whose name ends in `_file` to accept the filename version separately from the unsuffixed direct value version.

Agreed on not making a breaking change to the existing argument right now, but I suggest following the new pattern for the new argument, at the expense of consistency, and planning to break this _eventually_ to make the older argument consistent with the new design at a later date.

Reply:
That's useful background info about it being a shim, in that case I'll avoid using it for the new field like you suggested. 

If we eventually break the `encryption_key` field and add `encryption_key_file` (or equiv.) I'll avoid updating the documentation in this PR;  we probably don't want to encourage more people to creating configurations that would experience the breaking change!